File: src/model/Biscuit.java
package model;

public class Biscuit {

    private BiscuitType biscuitType;

    private BiscuitState biscuitState;

    public Biscuit(BiscuitType biscuitType) {
        this.biscuitType = biscuitType;
        this.biscuitState = BiscuitState.RAW;
    }

    public BiscuitState getBiscuitState() { return this.biscuitState; }

    public void setBiscuitState(BiscuitState biscuitState) {
        this.biscuitState = biscuitState;
    }
}

File: js-solution/components/biscuit.js
import { biscuitState } from "./biscuitState.js";

const biscuit = (flavor) => {

    let state = {
        flavor,
        biscuitState: biscuitState.RAW,
    };

    return {
        getBiscuitState: () => {
            return state.biscuitState;
        },
        setBiscuitState: (newState) => {
            state.biscuitState = newState;
        }
    }
};

export { biscuit };
